catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/endpoint/impl/CatalogEndpointImpl.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.endpoint.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.codice.ddf.configuration.PropertyResolver;
+
+import ddf.catalog.endpoint.CatalogEndpoint;
+
+public class CatalogEndpointImpl implements CatalogEndpoint {
+    protected Map<String, String> endpointProperties;
+
+    public static final String URL_BINDING_NAME_KEY = "urlBindingName";
+
+    /**
+     * Constructor
+     * <p>
+     * Creates empty endpoint properties map
+     */
+    public CatalogEndpointImpl() {
+        setEndpointProperties(new HashMap<>());
+    }
+
+    /**
+     * Constructor
+     *
+     * @param endpointProperties - a map of endpoint properties
+     */
+    public CatalogEndpointImpl(Map<String, String> endpointProperties) {
+        if (endpointProperties != null) {
```

Comment:
Same as below, I'd prefer a positive conditional.
```
if (endpointProperties == null) {
    setEndpointProperties(new HashMap<>());
} else {
    setEndpointProperties(endpointProperties);
}
```
People usually find them a bit easier to think about.